dali/image/tiff.cc

```diff
@@ -80,15 +81,20 @@ Image::Shape TiffImage::PeekShapeImpl(const uint8_t *encoded_buffer, size_t leng
       } else if (tag_id == HEIGHT_TAG) {
         height = value;
         height_read = true;
-      } else if (tag_id == SAMPLESPERPIXEL_TAG) {
+      } else if (tag_id == SAMPLESPERPIXEL_TAG && !photometric_read) {
         nchannels = value;
         nchannels_read = true;
+      } else if (tag_id == PHOTOMETRIC_INTERPRETATION_TAG && value == PHOTOMETRIC_PALETTE) {
+        nchannels = 3;
+        photometric_read = true;
       }
     }
+    if (width_read && height_read && nchannels_read && photometric_read)
+      break;
   }
 
-  DALI_ENFORCE(width_read && height_read && nchannels_read,
-    "TIFF image dims haven't been peeked properly");
+  DALI_ENFORCE(width_read && height_read && (nchannels_read | photometric_read),
```

Comment:
It's logical formula: 
```suggestion
  DALI_ENFORCE(width_read && height_read && (nchannels_read || photometric_read),
```

Reply:
Fixed, thanks!